modules/hostap/CMakeLists.txt

```diff
@@ -539,137 +539,138 @@ zephyr_library_compile_definitions_ifdef(CONFIG_EAP_FAST
-	${HOSTAP_SRC_BASE}/crypto/aes-wrap.c
-	${HOSTAP_SRC_BASE}/crypto/aes-unwrap.c
-	${HOSTAP_SRC_BASE}/crypto/rc4.c
-	${HOSTAP_SRC_BASE}/crypto/sha1-internal.c
-	${HOSTAP_SRC_BASE}/crypto/sha1-prf.c
-	${HOSTAP_SRC_BASE}/crypto/sha1-tlsprf.c
-	${HOSTAP_SRC_BASE}/crypto/sha256-prf.c
-	${HOSTAP_SRC_BASE}/crypto/sha256-kdf.c
-	${HOSTAP_SRC_BASE}/crypto/sha384-prf.c
-	${HOSTAP_SRC_BASE}/crypto/sha384-kdf.c
-	${HOSTAP_SRC_BASE}/crypto/sha512-internal.c
-	${HOSTAP_SRC_BASE}/crypto/sha512.c
-	${HOSTAP_SRC_BASE}/crypto/sha512-prf.c
-	${HOSTAP_SRC_BASE}/crypto/sha512-kdf.c
+  ${HOSTAP_SRC_BASE}/crypto/crypto_mbedtls-bignum.c
+  ${HOSTAP_SRC_BASE}/crypto/crypto_mbedtls-ec.c
+  ${HOSTAP_SRC_BASE}/crypto/crypto_mbedtls.c
+  ${HOSTAP_SRC_BASE}/crypto/aes-internal.c
+  ${HOSTAP_SRC_BASE}/crypto/aes-wrap.c
+  ${HOSTAP_SRC_BASE}/crypto/aes-unwrap.c
+  ${HOSTAP_SRC_BASE}/crypto/rc4.c
+  ${HOSTAP_SRC_BASE}/crypto/sha1-internal.c
+  ${HOSTAP_SRC_BASE}/crypto/sha1-prf.c
+  ${HOSTAP_SRC_BASE}/crypto/sha1-tlsprf.c
+  ${HOSTAP_SRC_BASE}/crypto/sha256-prf.c
+  ${HOSTAP_SRC_BASE}/crypto/sha256-kdf.c
+  ${HOSTAP_SRC_BASE}/crypto/sha384-prf.c
+  ${HOSTAP_SRC_BASE}/crypto/sha384-kdf.c
+  ${HOSTAP_SRC_BASE}/crypto/sha512-internal.c
+  ${HOSTAP_SRC_BASE}/crypto/sha512.c
+  ${HOSTAP_SRC_BASE}/crypto/sha512-prf.c
+  ${HOSTAP_SRC_BASE}/crypto/sha512-kdf.c
 )
 
 zephyr_library_sources_ifdef(CONFIG_WIFI_NM_WPA_SUPPLICANT_WPA3
-	${HOSTAP_SRC_BASE}/crypto/sha256-kdf.c
+  ${HOSTAP_SRC_BASE}/crypto/sha256-kdf.c
 )
 
 zephyr_library_sources_ifndef(CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ENTERPRISE
-	${HOSTAP_SRC_BASE}/crypto/tls_none.c
+  ${HOSTAP_SRC_BASE}/crypto/tls_none.c
 )
 
 zephyr_library_sources_ifdef(CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ENTERPRISE
-	# common
-	${HOSTAP_SRC_BASE}/crypto/sha384-tlsprf.c
-	${HOSTAP_SRC_BASE}/crypto/sha256-tlsprf.c
-	${HOSTAP_SRC_BASE}/crypto/sha1-tlsprf.c
-	${HOSTAP_SRC_BASE}/crypto/sha1-tprf.c
-	${HOSTAP_SRC_BASE}/crypto/ms_funcs.c
-	${HOSTAP_SRC_BASE}/crypto/aes-eax.c
-	# MD4 removed from MbedTLS
-	${HOSTAP_SRC_BASE}/crypto/md4-internal.c
-	${HOSTAP_SRC_BASE}/crypto/aes-encblock.c
-	${HOSTAP_SRC_BASE}/crypto/tls_mbedtls.c
+  # common
+  ${HOSTAP_SRC_BASE}/crypto/sha384-tlsprf.c
+  ${HOSTAP_SRC_BASE}/crypto/sha256-tlsprf.c
+  ${HOSTAP_SRC_BASE}/crypto/sha1-tlsprf.c
+  ${HOSTAP_SRC_BASE}/crypto/sha1-tprf.c
+  ${HOSTAP_SRC_BASE}/crypto/ms_funcs.c
+  ${HOSTAP_SRC_BASE}/crypto/aes-eax.c
+  # MD4 removed from MbedTLS
+  ${HOSTAP_SRC_BASE}/crypto/md4-internal.c
+  ${HOSTAP_SRC_BASE}/crypto/aes-encblock.c
+  ${HOSTAP_SRC_BASE}/crypto/tls_mbedtls.c
 )
 endif()
 
 if(CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT)
 zephyr_include_directories(
-	${HOSTAP_BASE}/port/mbedtls
+  ${HOSTAP_BASE}/port/mbedtls
 )
 
 zephyr_library_sources(
-	${HOSTAP_SRC_BASE}/crypto/crypto_mbedtls_alt.c
-	${HOSTAP_SRC_BASE}/crypto/tls_mbedtls_alt.c
-	${HOSTAP_SRC_BASE}/crypto/rc4.c
+  ${HOSTAP_SRC_BASE}/crypto/crypto_mbedtls_alt.c
+  ${HOSTAP_SRC_BASE}/crypto/tls_none.c
```

Comment:
will tls_none.c conflict with tls_mbedtls_alt.c when
CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ALT=y and
CONFIG_WIFI_NM_WPA_SUPPLICANT_CRYPTO_ENTERPRISE=y?

Reply:
I have fixed the conflicts, `tls_none.c` is only if Enterprise isn't selected.